### PR TITLE
npir using generic nodes

### PR DESCRIPTION
**Pull request by DropD**
## Description

Make NumpyIR use generic nodes.

### CODE REVIEW

File: src/gt4py/gtc/python/npir.py [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
+class Expr(common.Expr):
+    # MARK: remove when abstract nodes implemented in eve
+    def __init__(self, *args, **kwargs):
+        if type(self) is Expr:
+            raise TypeError("Cannot instantiate abstract Expr type of numpy IR")
+        super().__init__(*args, **kwargs)
+
+
+class Literal(common.Literal, Expr):
+    kind = cast(common.ExprKind, common.ExprKind.SCALAR)
```
Comment (havogt) [inline src/gt4py/gtc/python/npir.py:18]: What's this cast doing?
Compared to 
```
    kind: common.ExprKind = common.ExprKind.SCALAR
```

File: src/gt4py/gtc/python/npir.py [hunk 2]
```diff
@@ -47,8 +57,14 @@ def k(cls, offset: int, *, parallel=False) -> "AxisOffset":
         return cls.from_int(axis_name=AxisName.K, offset=offset, parallel=parallel)
 
 
-class VectorExpression(gtir.LocNode):
-    pass
+class VectorExpression(Expr):
+    # MARK: remove when abstract nodes implemented in eve
```
Comment (havogt) [inline src/gt4py/gtc/python/npir.py:61]: Why `MARK`?

File: src/gt4py/gtc/python/npir.py (hunk 1 above)
Comment (DropD) [reply src/gt4py/gtc/python/npir.py:18]: It simply tells mypy that this is an ExprKind instance, not an int (it gets confused for some reason). At runtime it does nothing.

File: src/gt4py/gtc/python/npir.py (hunk 2 above)
Comment (DropD) [reply src/gt4py/gtc/python/npir.py:61]: because it might never get done but when it happens it is easy to search & replace for

Comment (DropD) [reply src/gt4py/gtc/python/npir.py:61]: will change it to `TODO` to stick with the current convention.

**Pull request merged**